src/main/flight/pid.c

```diff
@@ -771,14 +774,32 @@ void FAST_CODE pidController(const pidProfile_t *pidProfile, const rollAndPitchT
 
     const float tpaFactor = getThrottlePIDAttenuation();
     const float motorMixRange = getMotorMixRange();
+    const float throttleHpf = getThrottleHpf();
 
 #ifdef USE_YAW_SPIN_RECOVERY
     const bool yawSpinActive = gyroYawSpinDetected();
 #endif
 
     // Dynamic i component,
     // gradually scale back integration when above windup point
-    const float dynCi = MIN((1.0f - motorMixRange) * ITermWindupPointInv, 1.0f) * dT * itermAccelerator;
+    float dynCi;
+#if defined(USE_THROTTLE_BOOST)
+    if (antiGravityNew && (throttleBoost > 0)) {
+    // Calculate new smooth anti-gravity effect where I accumulates faster with faster throttle movement
+    // ThrottleHpf typically reaches max of 0.1 for fast throttle changes
+    // AG Gain of 5000 results in itermAccelerateNew of 5 at throttleHpf of 0.1
+    // if throttle movement is zero, throttleHpf is zero, I accumulates normally
+        float itermAccelerateNew = (1 + (fabsf(throttleHpf) * 0.01 * (pidProfile->itermAcceleratorGain - 1000)));
```

Comment:
Is there any need for negative gain? And even in that case itermAcceleratorGain shall be changes to signed, not offset by 1000 ...

Reply:
The range for gain in CLI is 1000 to some much bigger number.  1000 translates to a gain of 1.  In order to retain the old gain values, the subtraction was necessary.  The output of that line should be 1 when the user sets the gain to 1.